### PR TITLE
app-mob status

### PR DESCRIPTION
**Pull request by abhi16394**
# Description
Adding Application mobility status field and also checking services dependent on application-mobility if its succeeded in deploying
https://jira.cec.lab.emc.com/browse/KRV-15319
# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration
We can see the status field as succeeded
image

### CODE REVIEW

File: pkg/utils/status.go [hunk 1]
```diff
@@ -273,11 +279,17 @@ func calculateState(ctx context.Context, instance *csmv1.ContainerStorageModule,
 func SetStatus(ctx context.Context, r ReconcileCSM, instance *csmv1.ContainerStorageModule, newStatus *csmv1.ContainerStorageModuleStatus) {
 
 	log := logger.GetLogger(ctx)
-	instance.GetCSMStatus().State = newStatus.State
-	log.Infow("Driver State", "Controller",
-		newStatus.ControllerStatus, "Node", newStatus.NodeStatus)
-	instance.GetCSMStatus().ControllerStatus = newStatus.ControllerStatus
-	instance.GetCSMStatus().NodeStatus = newStatus.NodeStatus
+	if instance.GetName() == "application-mobility" {
```
Comment (kumarkgosa) [inline pkg/utils/status.go:282]: even though appmob is the only module to have csm state, wouldn't it be better to have a separate function to set status for appmob ?

**Review by kumarkgosa — APPROVED** [review]

File: pkg/utils/status.go (hunk 1 above)
Comment (abhi16394) [reply pkg/utils/status.go:282]: if there are more modules i was thinking of having a different function.
That if for future update

**Review by mbasha-dell — APPROVED** [review]

**Review by jooseppi-luna — CHANGES_REQUESTED** [review]
have you successfully run app-mob e2e tests?

File: operatorconfig/moduleconfig/application-mobility/v0.3.0/velero-deployment.yaml [hunk 2]
```diff
@@ -172,13 +174,18 @@ spec:
...
+           - mountPath: /target
+             name: plugins
+        - name: <DELL_INIT_CONTAINER_NAME>
+          image: <DELL_INIT_CONTAINER_IMAGE>
           volumeMounts:
            - mountPath: /target
              name: plugins
       volumes:
-        - name: <APPLICATION_MOBILITY_OBJECT_STORE_SECRET_NAME>
+        - name: cloud-credentials
```
[the remark this answers is not in the document]
Comment (kumarkgosa) [reply operatorconfig/moduleconfig/application-mobility/v0.3.0/velero-deployment.yaml:188]: Volume name was never configurable, only the secret name was. We can name volume anything (not to confuse with secret name). Hence, maintaining consistency from csm-app-mob.

[the remark this answers is not in the document]
Comment (jooseppi-luna) [reply operatorconfig/moduleconfig/application-mobility/v0.3.0/velero-deployment.yaml:188]: oh ok perfect thanks for the explanation

**Pull request closed without merging**